Runtime/Dart/lib/src/bebop.dart

```diff
@@ -153,19 +151,28 @@ class BebopWriter {
   ByteData _view;
   int length = 0;
 
-static const Utf8Encoder _utf8Encoder = Utf8Encoder();
+  static const Utf8Encoder _utf8Encoder = Utf8Encoder();
 
   BebopWriter._() {
     _buffer = _bytes.buffer;
     _view = ByteData.view(_buffer);
   }
 
   static final BebopWriter _instance = BebopWriter._();
-  factory BebopWriter() => _instance..length = 0;
+  factory BebopWriter() {
+    // clear buffer -> enables stacked byte schemes
+    _instance._bytes = Uint8List(256);
+    _instance._buffer = _instance._bytes.buffer;
+    _instance._view = ByteData.view(_instance._buffer);
+    _instance.length = 0;
+    return _instance;
+  }
 
   void _guaranteeBufferLength(int length) {
     if (length > _bytes.lengthInBytes) {
-      final data = Uint8List(min(2 * _bytes.lengthInBytes, length));
+      // ensure data is big enough if we deal with huge buffers
+      final data =
+          Uint8List(min(max(2 * _bytes.lengthInBytes, length), length));
```

Comment:
You know, thinking it over, I think this just wants to be `max(2 * _bytes.lengthInBytes, length)`. I probably mixed up `max` and `min` in the original line.

The idea here is to double the old array size, because otherwise you have to resize-and-copy the whole array with every push. So if the old size is 64 and `_guaranteeBufferLength(68)` is called, then we immediately resize to 128.

But of course if a huge buffer is pushed all at once, like `_guaranteeBufferLength(999)`, then doubling to 128 isn't enough, and we should resize to the 999 asked for. I assume this is what your comment here alludes to.

However, this new formula `min(max(2 * _bytes.lengthInBytes, length), length)` is always equal to `length`.

I think that `max(2 * _bytes.lengthInBytes, length)` is correct.

Reply:
True! Wasn't thinking to much about it :D